exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -76,6 +76,14 @@ func (m *AzureManagedMachinePool) ValidateUpdate(oldRaw runtime.Object, client c
 	old := oldRaw.(*AzureManagedMachinePool)
 	var allErrs field.ErrorList
 
+	if m.Spec.OSType != old.Spec.OSType {
```

Comment:
`OSType` is a `*string`, so you'll have to follow the pattern from `OSDiskSizeGB` below, e.g.:

```
	if old.Spec.OSType != nil {
		// Prevent OSType removal if it was already set to some value
		if m.Spec.OSType == nil {
			// unsetting the field is not allowed
			allErrs = append(allErrs,
				field.Invalid(
					field.NewPath("Spec", "OSType"),
					m.Spec.OSType,
					"field is immutable, unsetting is not allowed"))
		} else if *m.Spec.OSType != *old.Spec.OSType {
			// changing the field is not allowed
			allErrs = append(allErrs,
				field.Invalid(
					field.NewPath("Spec", "OSType"),
					*m.Spec.OSType,
					"field is immutable"))
		}
	}
```

Reply:
moof... my bad

Reply:
fixed!